Add executeWithRetry when running the test matrix

- https://github.com/TestArmada/flank/blob/526694c7eb39a85cc721ce8d483121b55162de4f/test_runner/src/main/kotlin/ftl/run/IosTestRunner.kt#L56
- https://github.com/TestArmada/flank/blob/526694c7eb39a85cc721ce8d483121b55162de4f/test_runner/src/main/kotlin/ftl/run/AndroidTestRunner.kt#L46

- https://github.com/TestArmada/flank/blob/86a530f0778f8ce9f8c260f852b1524c767d6c5a/test_runner/src/main/kotlin/ftl/http/ExecuteWithRetry.kt#L8

```
15:02:04 Exception in thread "main" picocli.CommandLine$ExecutionException: Error while running command (ftl.cli.firebase.test.android.AndroidRunCommand@78d39a69): com.google.api.client.googleapis.json.GoogleJsonResponseException: 500 Internal Server Error
15:02:04 {
15:02:04   "code" : 500,
15:02:04   "errors" : [ {
15:02:04     "domain" : "global",
15:02:04     "message" : "Internal error encountered.",
15:02:04     "reason" : "backendError"
15:02:04   } ],
15:02:04   "message" : "Internal error encountered.",
15:02:04   "status" : "INTERNAL"
15:02:04 }
15:02:04    at picocli.CommandLine.execute(CommandLine.java:1168)
15:02:04    at picocli.CommandLine.access$800(CommandLine.java:141)
15:02:04    at picocli.CommandLine$RunLast.handle(CommandLine.java:1367)
15:02:04    at picocli.CommandLine$RunLast.handle(CommandLine.java:1335)
15:02:04    at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1243)
15:02:04    at picocli.CommandLine.parseWithHandlers(CommandLine.java:1526)
15:02:04    at picocli.CommandLine.run(CommandLine.java:1974)
15:02:04    at picocli.CommandLine.run(CommandLine.java:1920)
15:02:04    at ftl.Main$Companion.main(Main.kt:48)
15:02:04    at ftl.Main.main(Main.kt)
15:02:04 Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 500 Internal Server Error
15:02:04 {
15:02:04   "code" : 500,
15:02:04   "errors" : [ {
15:02:04     "domain" : "global",
15:02:04     "message" : "Internal error encountered.",
15:02:04     "reason" : "backendError"
15:02:04   } ],
15:02:04   "message" : "Internal error encountered.",
15:02:04   "status" : "INTERNAL"
15:02:04 }
15:02:04    at com.google.api.client.googleapis.json.GoogleJsonResponseException.from(GoogleJsonResponseException.java:150)
15:02:04    at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:113)
15:02:04    at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:40)
15:02:04    at com.google.api.client.googleapis.services.AbstractGoogleClientRequest$1.interceptResponse(AbstractGoogleClientRequest.java:401)
15:02:04    at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1097)
15:02:04    at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:499)
15:02:04    at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:432)
15:02:04    at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:549)
15:02:04    at ftl.run.AndroidTestRunner$runTests$2$invokeSuspend$$inlined$repeat$lambda$1.invokeSuspend(AndroidTestRunner.kt:46)
15:02:04    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:32)
15:02:04    at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:233)
15:02:04    at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.kt:116)
15:02:04    at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:76)
15:02:04    at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:53)
15:02:04    at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
15:02:04    at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:35)
15:02:04    at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
15:02:04    at ftl.cli.firebase.test.android.AndroidRunCommand.run(AndroidRunCommand.kt:34)
15:02:04    at picocli.CommandLine.execute(CommandLine.java:1160)
15:02:04    ... 9 more
```

--- a/test_runner/src/main/kotlin/ftl/run/AndroidTestRunner.kt
+++ b/test_runner/src/main/kotlin/ftl/run/AndroidTestRunner.kt
@@ -7,6 +7,7 @@ import ftl.gc.GcAndroidDevice
 import ftl.gc.GcAndroidTestMatrix
 import ftl.gc.GcStorage
 import ftl.gc.GcToolResults
+import ftl.http.executeWithRetry
 import ftl.json.MatrixMap
 import ftl.util.ShardCounter
 import kotlinx.coroutines.Deferred
@@ -43,7 +44,7 @@ object AndroidTestRunner {
                         args = androidArgs,
                         shardCounter = shardCounter,
                         toolResultsHistory = history
-                    ).execute()
+                    ).executeWithRetry()
                 }
             }
         }

--- a/test_runner/src/main/kotlin/ftl/run/IosTestRunner.kt
+++ b/test_runner/src/main/kotlin/ftl/run/IosTestRunner.kt
@@ -7,6 +7,7 @@ import ftl.gc.GcIosMatrix
 import ftl.gc.GcIosTestMatrix
 import ftl.gc.GcStorage
 import ftl.gc.GcToolResults
+import ftl.http.executeWithRetry
 import ftl.ios.Xctestrun
 import ftl.json.MatrixMap
 import ftl.run.GenericTestRunner.beforeRunMessage
@@ -53,7 +54,7 @@ object IosTestRunner {
                         args = iosArgs,
                         shardCounter = shardCounter,
                         toolResultsHistory = history
-                    ).execute()
+                    ).executeWithRetry()
                 }
             }
         }

